File: app/src/main/java/com/joyhonest/sports_camera/Storage.java
package com.joyhonest.sports_camera;

import android.content.Context;
import android.os.Environment;
import android.os.StatFs;

import java.io.BufferedReader;
import java.io.File;
import java.io.FileNotFoundException;
import java.io.IOException;
import java.io.InputStreamReader;

import tv.danmaku.ijk.media.player.IjkMediaMeta;
public class Storage {
    public static final String VENDOR = "JH-WIFI-FPV";

    public static String getPhoneCardPath() {
        return Environment.getDataDirectory().getPath();
    }

    public static String getNormalSDCardPath() {
        return Environment.getExternalStorageDirectory().getPath();
    }

    public static String getSDCardPathEx() {
        String str = new String();
        try {
            BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(Runtime.getRuntime().exec("mount").getInputStream()));
            while (true) {
                String readLine = bufferedReader.readLine();
                if (readLine == null) {
                    break;
                } else if (!readLine.contains("secure") && !readLine.contains("asec") && (readLine.contains("fat") || readLine.contains("fuse"))) {
                    String[] split = readLine.split(" ");
                    if (split != null && split.length > 1 && getTotalSize(split[1]) > IjkMediaMeta.AV_CH_STEREO_RIGHT && !getNormalSDCardPath().equals(split[1])) {
                        str = str.concat(split[1] + ";");
                    }
                }
            }
        } catch (FileNotFoundException e) {
            e.printStackTrace();
        } catch (IOException e2) {
            e2.printStackTrace();
        }
        return str;
    }

    public static long getTotalSize(String str) {
        try {
            StatFs statFs = new StatFs(new File(str).getPath());
            return statFs.getBlockSizeLong() * statFs.getBlockCountLong();
        } catch (Exception e) {
            e.printStackTrace();
            return 0L;
        }
    }

    public static long getAvailableSize(String str) {
        try {
            StatFs statFs = new StatFs(new File(str).getPath());
            long blockSizeLong = statFs.getBlockSizeLong() * statFs.getAvailableBlocksLong();
            statFs.getBlockSizeLong();
            statFs.getBlockCountLong();
            return blockSizeLong;
        } catch (Exception e) {
            e.printStackTrace();
            return 0L;
        }
    }

    public static int dip2px(Context context, float f) {
        return (int) ((f * context.getResources().getDisplayMetrics().density) + 0.5f);
    }

    public static float dip2pxF(Context context, float f) {
        return f * context.getResources().getDisplayMetrics().density;
    }

    public static int px2dip(Context context, float f) {
        return (int) ((f / context.getResources().getDisplayMetrics().density) + 0.5f);
    }
}

File: app/src/main/java/com/joyhonest/sports_camera/Storage.kt
package com.joyhonest.sports_camera

import android.content.Context
import android.os.Environment
import android.os.StatFs
import tv.danmaku.ijk.media.player.IjkMediaMeta
import java.io.BufferedReader
import java.io.File
import java.io.FileNotFoundException
import java.io.IOException
import java.io.InputStreamReader

object Storage {
    const val VENDOR = "JH-WIFI-FPV"
    val phoneCardPath: String
        get() = Environment.getDataDirectory().path
    val normalSDCardPath: String
        get() = Environment.getExternalStorageDirectory().path
    val sDCardPathEx: String
        get() {
            var str = String()
            try {
                val bufferedReader = BufferedReader(InputStreamReader(Runtime.getRuntime().exec("mount").inputStream))
                while (true) {
                    val readLine = bufferedReader.readLine()
                    if (readLine == null) {
                        break
                    } else if (!readLine.contains("secure") && !readLine.contains("asec") && (readLine.contains("fat") || readLine.contains("fuse"))) {
                        val split = readLine.split(" ".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
                        if (split != null && split.size > 1 && getTotalSize(split[1]) > IjkMediaMeta.AV_CH_STEREO_RIGHT && normalSDCardPath != split[1]) {
                            str = str + split[1] + ";"
                        }
                    }
                }
            } catch (e: FileNotFoundException) {
                e.printStackTrace()
            } catch (e2: IOException) {
                e2.printStackTrace()
            }
            return str
        }

    fun getTotalSize(str: String?): Long {
        return try {
            val statFs = StatFs(File(str).path)
            statFs.blockSizeLong * statFs.blockCountLong
        } catch (e: Exception) {
            e.printStackTrace()
            0L
        }
    }

    fun getAvailableSize(str: String?): Long {
        return try {
            val statFs = StatFs(File(str).path)
            val blockSizeLong = statFs.blockSizeLong * statFs.availableBlocksLong
            statFs.blockSizeLong
            statFs.blockCountLong
            blockSizeLong
        } catch (e: Exception) {
            e.printStackTrace()
            0L
        }
    }

    fun dip2px(context: Context, f: Float): Int {
        return (f * context.resources.displayMetrics.density + 0.5f).toInt()
    }

    fun dip2pxF(context: Context, f: Float): Float {
        return f * context.resources.displayMetrics.density
    }

    fun px2dip(context: Context, f: Float): Int {
        return (f / context.resources.displayMetrics.density + 0.5f).toInt()
    }
}